tests/certification/pubsub/azure/servicebus/servicebus_test.go

```diff
@@ -0,0 +1,1136 @@
+				pubsub_loader.New("azure.servicebus", func() pubsub.PubSub {
+					return pubsub_servicebus.NewAzureServiceBus(log)
+				}),
+			),
+			runtime.WithSecretStores(secretstores_loader.New("local.env", func() secretstores.SecretStore {
+				return secretstore_env.NewEnvSecretStore(log)
+			})))
+		sidecar.Start(ctx)
+		return nil
+	}
+
+	flow.New(t, "servicebus certification - entity management enabled").
+
+		// Test : disableEntityManagement optional attribute , Test processing with all optional attributes and single publisher/subscriber
+		// Run subscriberApplication app1
+		Step("verify if app1 has recevied messages published to newly created topic", testDefault).
+		Step(fmt.Sprintf("publish messages to topicDefault: %s", topicDefaultName), publishMessages(metadata, sidecarName1, topicDefaultName, consumerGroup1)).
+		Run()
+}
+
+func TestServicebusDefaultTtl(t *testing.T) {
+	log := logger.NewLogger("dapr.components")
+	component := pubsub_loader.New("azure.servicebus", func() pubsub.PubSub {
+		return pubsub_servicebus.NewAzureServiceBus(log)
+	})
+
+	secretStoreComponent := secretstores_loader.New("local.env", func() secretstores.SecretStore {
+		return secretstore_env.NewEnvSecretStore(log)
+	})
+
+	consumerGroup1 := watcher.NewUnordered()
+
+	// Set the partition key on all messages so they are written to the same partition. This allows for checking of ordered messages.
+	metadata := map[string]string{
+		messageKey: partition0,
+	}
+
+	// subscriber of the given topic
+	subscriberApplication := func(appID string, topicName string, messagesWatcher *watcher.Watcher) app.SetupFn {
+		return func(ctx flow.Context, s common.Service) error {
+			// Simulate periodic errors.
+			sim := simulate.PeriodicError(ctx, 100)
+			// Setup the /orders event handler.
+			return multierr.Combine(
+				s.AddTopicEventHandler(&common.Subscription{
+					PubsubName: pubsubName,
+					Topic:      topicName,
+					Route:      "/orders",
+				}, func(_ context.Context, e *common.TopicEvent) (retry bool, err error) {
+					if err := sim(); err != nil {
+						return true, err
+					}
+
+					// Track/Observe the data of the event.
+					messagesWatcher.Observe(e.Data)
+					ctx.Logf("Message Received appID: %s,pubsub: %s, topic: %s, id: %s, data: %s", appID, e.PubsubName, e.Topic, e.ID, e.Data)
+					return false, nil
+				}),
+			)
+		}
+	}
+
+	testTtlPublishMessages := func(metadata map[string]string, sidecarName string, topicName string, messageWatchers ...*watcher.Watcher) flow.Runnable {
+		return func(ctx flow.Context) error {
+			// prepare the messages
+			messages := make([]string, numMessages)
+			for i := range messages {
+				messages[i] = fmt.Sprintf("partitionKey: %s, message for topic: %s, index: %03d, uniqueId: %s", metadata[messageKey], topicName, i, uuid.New().String())
+			}
+
+			// add the messages as expectations to the watchers
+			for _, messageWatcher := range messageWatchers {
+				messageWatcher.ExpectStrings(messages...)
```

Comment:
You are adding all messages to the message watcher.. even though you want to check that all (or some) of the messages are not sent because they expire.

So maybe do *not* add the message here.